File: src/main/java/setting/SettingsConfigurable.java
package setting;

import com.intellij.openapi.options.Configurable;
import helper.DataService;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class SettingsConfigurable implements Configurable {

    private final DataService data = DataService.getInstance();
    private SettingsComponent mSetting;

    @Nls(capitalization = Nls.Capitalization.Title)
    @Override
    public String getDisplayName() {
        return "GetX Setting";
    }

    @Nullable
    @Override
    public JComponent createComponent() {
        mSetting = new SettingsComponent();
        return mSetting.mainPanel;
    }

    @Override
    public boolean isModified() {
        boolean modified;
        modified = !mSetting.logicName.getText().equals(data.logicName)
                || !mSetting.stateName.getText().equals(data.stateName)
                || !mSetting.viewName.getText().equals(data.viewName)
                || !mSetting.viewFileName.getText().equals(data.viewFileName);
        return modified;
    }

    @Override
    public void apply() {
        data.logicName = mSetting.logicName.getText();
        data.stateName = mSetting.stateName.getText();
        data.viewName = mSetting.viewName.getText();
        data.viewFileName = mSetting.viewFileName.getText();
    }

    @Override
    public void reset() {
        mSetting.logicName.setText(data.logicName);
        mSetting.stateName.setText(data.stateName);
        mSetting.viewName.setText(data.viewName);
        mSetting.viewFileName.setText(data.viewFileName);
    }

    @Override
    public void disposeUIResources() {
        mSetting = null;
    }
}

File: src/main/kotlin/setting/SettingsConfigurable.kt
package setting

import com.intellij.openapi.options.Configurable
import helper.DataService
import org.jetbrains.annotations.Nls
import javax.swing.JComponent

class SettingsConfigurable : Configurable {
    private val data = DataService.instance
    private var mSetting: SettingsComponent? = null

    @Nls(capitalization = Nls.Capitalization.Title)
    override fun getDisplayName(): String {
        return "GetX Setting"
    }

    override fun createComponent(): JComponent {
        mSetting = SettingsComponent()
        return mSetting!!.mainPanel
    }

    override fun isModified(): Boolean {
        return (//Page
                mSetting!!.page.logic.text != data.templatePage.logic
                        || mSetting!!.page.state.text != data.templatePage.state
                        || mSetting!!.page.view.text != data.templatePage.view
                        || mSetting!!.page.viewFile.text != data.templatePage.viewFile
                        //Component
                        || mSetting!!.component.logic.text != data.templateComponent.logic
                        || mSetting!!.component.state.text != data.templateComponent.state
                        || mSetting!!.component.view.text != data.templateComponent.view
                        || mSetting!!.component.viewFile.text != data.templateComponent.viewFile
                        //Custom
                        || mSetting!!.custom.logic.text != data.templateCustom.logic
                        || mSetting!!.custom.state.text != data.templateCustom.state
                        || mSetting!!.custom.view.text != data.templateCustom.view
                        || mSetting!!.custom.viewFile.text != data.templateCustom.viewFile
                )
    }

    override fun apply() {
        //Page
        data.templatePage.logic = mSetting!!.page.logic.text
        data.templatePage.state = mSetting!!.page.state.text
        data.templatePage.view = mSetting!!.page.view.text
        data.templatePage.viewFile = mSetting!!.page.viewFile.text
        //Component
        data.templateComponent.logic = mSetting!!.component.logic.text
        data.templateComponent.state = mSetting!!.component.state.text
        data.templateComponent.view = mSetting!!.component.view.text
        data.templateComponent.viewFile = mSetting!!.component.viewFile.text
        //Custom
        data.templateCustom.logic = mSetting!!.custom.logic.text
        data.templateCustom.state = mSetting!!.custom.state.text
        data.templateCustom.view = mSetting!!.custom.view.text
        data.templateCustom.viewFile = mSetting!!.custom.viewFile.text
    }

    override fun reset() {
        //page
        mSetting!!.page.logic.text = data.templatePage.logic
        mSetting!!.page.state.text = data.templatePage.state
        mSetting!!.page.view.text = data.templatePage.view
        mSetting!!.page.viewFile.text = data.templatePage.viewFile
        //component
        mSetting!!.component.logic.text = data.templateComponent.logic
        mSetting!!.component.state.text = data.templateComponent.state
        mSetting!!.component.view.text = data.templateComponent.view
        mSetting!!.component.viewFile.text = data.templateComponent.viewFile
        //custom
        mSetting!!.custom.logic.text = data.templateCustom.logic
        mSetting!!.custom.state.text = data.templateCustom.state
        mSetting!!.custom.view.text = data.templateCustom.view
        mSetting!!.custom.viewFile.text = data.templateCustom.viewFile
    }

    override fun disposeUIResources() {
        mSetting = null
    }
}